processor/src/test/java/org/mapstruct/ap/test/java8stream/wildcard/WildCardTest.java

```diff
@@ -0,0 +1,134 @@
+/**
+ *  Copyright 2012-2016 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.java8stream.wildcard;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mapstruct.ap.testutil.IssueKey;
+import org.mapstruct.ap.testutil.WithClasses;
+import org.mapstruct.ap.testutil.compilation.annotation.CompilationResult;
+import org.mapstruct.ap.testutil.compilation.annotation.Diagnostic;
+import org.mapstruct.ap.testutil.compilation.annotation.ExpectedCompilationOutcome;
+import org.mapstruct.ap.testutil.runner.AnnotationProcessorTestRunner;
+
+/**
+ * @author Filip Hrisafov
+ */
+@IssueKey("962")
+@RunWith(AnnotationProcessorTestRunner.class)
+public class WildCardTest {
+
+    @Test
+    @WithClasses({
+        ExtendsBoundSourceTargetMapper.class,
+        ExtendsBoundSource.class,
+        Target.class,
+        Plan.class,
+        Idea.class
+    })
+    public void shouldGenerateExtendsBoundSourceForgedStreamMethod() {
```

Comment:
Minor thing, I'm seeing a FQN in the generated code here:

    protected List<Plan> ideaStreamToPlanList(Stream<? extends org.mapstruct.ap.test.java8stream.wildcard.Idea> {
        ...
    }